### DISCUSSION TITLE
Form Submit  with new fields to the same form getting filled with previous input data

### DISCUSSION THREAD

**Discussion by surya-trv-13** [react-hook-form/react-hook-form#5024] [Q&A] [answered]
I am creating a react app where the fields data is coming from the 
Here is my code sandbox: https://codesandbox.io/s/jovial-snowflake-jxqlj

The issue is when I am filling the form on page 1 with data and submit it loads the next page but the data on the second page gets auto-filled with the previous page input data (regardless of unique field names).

Here is few screenshots:

<img width="373" alt="FormPage1" src="https://user-images.githubusercontent.com/45998114/116965580-cb0e9080-accb-11eb-9f19-0d53e297dc13.PNG">

After Submit: (Clicking on Next):

<img width="374" alt="FormPage2" src="https://user-images.githubusercontent.com/45998114/116965646-ebd6e600-accb-11eb-9dac-e4d82495dcf9.PNG">

The same issue is also happening for the useFieldArray Id

Now Is there anything I can do to resolve the issue?

**Comment by bluebill1049** [comment, marked as answer]
you should use unique `key`

```diff
-key={index}
+key={formField.name + index}
```
